torch_geometric/nn/dense/dense_dmon_pool.py

```diff
@@ -0,0 +1,150 @@
+import torch
+from torch.nn import SELU, Dropout, Sequential
+
+from torch_geometric.nn.conv import MessagePassing
+from torch_geometric.nn.dense.linear import Linear
+
+EPS = 1e-15
+
+
+class DMonPooling(MessagePassing):
+    r"""Spectral modularity pooling operator from the `"Graph Clustering with
+    Graph Neural Networks" <https://arxiv.org/abs/2006.16904>`_ paper
+
+    .. math::
+        \mathbf{X}^{\prime} &= {\mathrm{softmax}(\mathbf{S})}^{\top} \cdot
+        \mathbf{X}
+
+        \mathbf{A}^{\prime} &= {\mathrm{softmax}(\mathbf{S})}^{\top} \cdot
+        \mathbf{A} \cdot \mathrm{softmax}(\mathbf{S})
+
+    based on dense learned assignments :math:`\mathbf{S} \in \mathbb{R}^{B
+    \times N \times C}`.
+    Returns learned cluster assignment matrix, pooled node feature matrix,
+    coarsened symmetrically normalized adjacency matrix, and two auxiliary
+    objectives: (1) The spectral loss
+
+    .. math::
+        \mathcal{L}_s = - \frac{1}{2m}
+        \cdot{\mathrm{Tr}(\mathbf{S}^{\top} \mathbf{B} \mathbf{S})}
+
+    where :math:`\mathbf{B}` is the modularity matrix, (2) the orthogonality
+    loss
+
+    .. math::
+        \mathcal{L}_o = {\left\| \frac{\mathbf{S}^{\top} \mathbf{S}}
+        {{\|\mathbf{S}^{\top} \mathbf{S}\|}_F} -\frac{\mathbf{I}_C}{\sqrt{C}}
+        \right\|}_F
+
+    where :math:`C` is the number of clusters, and (3) the cluster loss
+
+    .. math::
+        \mathcal{L}_c = \frac{\sqrt{C}}{n}
+        {\left\|\sum_i\mathbf{C_i}^{\top}\right\|}_F - 1.
+
+    Args:
+        dense_size (list): A 2D list whose each 1D list contains size for each
+            Linear layer being used.
+        k (int, optional): Number of clusters.
+            (default: :obj:`16`)
+        p (float, optional): Probability value for dropout layer.
+            (default: :obj:`0.2`)
+    """
+    def __init__(self, dense_size, k=16, p=0.2):
+        super().__init__()
+        self.dense_size = dense_size
+        self.k = k
+        self.dropout_rate = p
+
+        mlp = []
```

Comment:
Any chance we can use `torch_geometric.nn.MLP` for this?

Reply:
Unfortunately, it is required to apply **_Dropout_** to only **_Dense_** layer being used to compute soft cluster assignment matrix which `torch_geometric.nn.MLP` currently skips. Therefore, this method seems of no use to us. 

In essence, if we wanna use only one Dense layer, then, in the `forward` function of `torch_geometric.nn.MLP`, **_Dropout_** is gonna skip through as the length of `channel_list` is 2, which in turn, makes only one Dense layer. So, don't think we should use this method. Please comment on this if you've some suggestions on whether to use this method.......

Reply:
Can't you use `MLP` and apply a final `Dropout` layer afterwards?